src/libcollections/str.rs

```diff
@@ -10,12 +10,15 @@
 //
 // ignore-lexer-test FIXME #15679
 
-//! Unicode string manipulation (the `str` type).
+//! Unicode string manipulation (the [`str`](../primitive.str.html) type).
 //!
-//! Rust's `str` type is one of the core primitive types of the language. `&str` is the borrowed
-//! string type. This type of string can only be created from other strings, unless it is a static
-//! string (see below). As the word "borrowed" implies, this type of string is owned elsewhere, and
-//! this string cannot be moved out of.
+//! Rust's [`str`](../primitive.str.html) type is one of the core primitive types of the language.
+//! `&str` is the borrowed string type. This type of string can only be created from other strings,
+//! unless it is a static string (see below). As the word "borrowed" implies, this type of string
+//! is owned elsewhere, and this string cannot be moved out of.
```

Comment:
This pull request was merely correcting the issue, and I didn't touch any of the previous wording.  I agree that the earlier sentence feels a bit wonky, but I don't know if the dangling preposition is the cause.  IMO, this sentence flows much more naturally:
"It is not possible to move out of borrowed strings because they are owned elsewhere."  
It's not as ornate as the original language, but it is precise and concise.

In the end though, it's not very relevant to the original issue/reason for the pull request.  Not saying it's not worth discussion or changing, just that it's likely done lateral to this.